lib/metasploit/framework/data_service/remote/http/remote_credential_data_service.rb

```diff
@@ -10,15 +10,16 @@ module RemoteCredentialDataService
   def creds(opts = {})
     data = self.get_data(CREDENTIAL_API_PATH, nil, opts)
     rv = json_to_mdm_object(data, CREDENTIAL_MDM_CLASS, [])
-    parsed_body = JSON.parse(data.response.body)
-    parsed_body.each do |cred|
-      if cred['private']
-        private_object = to_ar(cred['private']['type'].constantize, cred['private'])
-        rv[parsed_body.index(cred)].private = private_object
+    parsed_body = JSON.parse(data.response.body).deep_symbolize_keys
```

Comment:
Why not pass `symbolize_names: true` to `JSON.parse` method?

Reply:
So looks like they behave differently. `symbolize_names: true` does convert every single key to a symbol:
`"{:data=>[{:id=>1, :name=>"default", :created_at=>"2018-07-23T17:12:02.745Z", :updated_at=>"2018-07-23T17:12:02.745Z", :boundary=>nil, :description=>nil, :owner_id=>nil, :limit_to_network=>false, :import_fingerprint=>false}]}"`

But `symbolize_keys` only does the top level hash
`"{:data=>[{"id"=>1, "name"=>"default", "created_at"=>"2018-07-23T17:12:02.745Z", "updated_at"=>"2018-07-23T17:12:02.745Z", "boundary"=>nil, "description"=>nil, "owner_id"=>nil, "limit_to_network"=>false, "import_fingerprint"=>false}]}"`

The `to_ar` method expects the input to have strings. Ideally it should work with symbols since I think we should have everything that uses parsed JSON expect symbols, but I'm not going to make that change with this PR.